custom/icds/forms.py

```diff
@@ -20,6 +24,20 @@
 from custom.icds_reports.const import CUSTOM_DATA_PULLS
 
 
+def validate_date(value):
+    error = forms.ValidationError(_("Please enter a valid date in the format YYYY-MM-DD"))
+
+    if not isinstance(value, (str, str)) or not re.match(r'^\d\d\d\d-\d\d-\d\d$', value):
```

Comment:
Yeah. The format is being controlled by UI, I guess we don't need a check on that. 
I searched around for methods for `validate_date` in HQ and found the example that I have used.
And the only reason I have used it was because I wanted to avoid try-catch block in `clean_*` methods which I feel was not required. 
@mkangia The snippet that you posted seems to do the job right and the `validate_date` can be removed from code.
Thanks for the suggestion 😄